frontend/packages/client/src/components/StepByStep/index.js

```diff
@@ -73,6 +64,96 @@ function StepByStep({
     [refs]
   );
 
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const goToStep = useCallback(
+    (step, replace = false) => {
+      searchParams.set('step', step + 1);
+      setSearchParams(searchParams, { replace });
+    },
+    [searchParams, setSearchParams]
+  );
+  const tryToGoForward = useCallback(() => {
+    const runPreCheckStepAdvance = () => {
+      if (refs.current) {
+        const runCheckResult = refs.current();
+        if (!runCheckResult) {
+          return false;
+        }
+        refs.current = null;
+      }
+      return true;
+    };
+    const canAdvance = runPreCheckStepAdvance();
+    if (currentStep + 1 < steps.length && canAdvance) {
+      goToStep(currentStep + 1);
+    }
+  }, [currentStep, goToStep, refs, steps.length]);
+  const tryToGoBack = useCallback(() => {
+    if (currentStep - 1 >= 0) {
+      goToStep(currentStep - 1);
+    }
+  }, [currentStep, goToStep]);
+
+  useEffect(() => {
+    // check if displayStep is valid
+    const displayStep = searchParams.get('step');
+    const displayStepNum = !isNaN(Number(displayStep))
+      ? Number(displayStep)
+      : -1;
+
+    if (showPreStep && displayStepNum <= 1) return;
+
+    if (
+      displayStep === null || // first time, need to set step in URL
+      displayStep === undefined || // first time, need to set step in URL
+      displayStepNum <= 0 || // prevent out-of-bounds values
+      displayStepNum > steps.length || // prevent out-of-bounds values
+      (showPreStep && displayStepNum > 1) // correct displayStep if preStep not dismissed
+    ) {
+      setValidSteps({});
+      goToStep(0, true);
+    }
+  }, [searchParams, steps.length, setValidSteps, goToStep, showPreStep]);
+
+  useEffect(() => {
+    if (showPreStep) return;
+
+    // check if valid condition to go forward/back
+    const displayStep = searchParams.get('step');
+    const displayStepNum = !isNaN(Number(displayStep))
+      ? Number(displayStep)
+      : -1;
+    const validStepsVals = Object.values(validSteps);
+    const numValidSteps = validStepsVals.length;
+    if (
+      (!preStep &&
+        // prevent reaching a subsequent step until prior steps have data
+        ((displayStepNum > 1 && (!numValidSteps || !allValidSteps())) ||
+          // or in the case of a mismatch
+          (!Object.keys(stepsData).length && numValidSteps))) ||
+      (preStep &&
+        ((displayStepNum > 2 && (!numValidSteps || !allValidSteps())) ||
+          (!Object.keys(stepsData).length && numValidSteps > 1)))
```

Comment:
This is causing an issue if there's a step that allows the user to not enter any data if it's not requited. 
I think we can remove checking if data is available as long as the step is valid. Data existence or validation is handled by each step and if it's ok then `isValidStep` is called. It can be tested on creating a community were step two adding admin and authors is empty it will cause to move to step 1 instead of step 3.